engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualizationJob.scala

```diff
@@ -157,6 +158,17 @@ class UpsertVisualizationJob(
 
 object UpsertVisualizationJob {
 
+  /** Invalidate caches for a particular expression id. */
+  sealed private case class InvalidateCaches(
+    expressionId: Api.ExpressionId
```

Comment:
case class will ensure that appropriate `equals` and `hashCode` are generated, and there are no unnecessary executions of the same logic before the program execution